packages/server/src/supertokens.ts

```diff
@@ -246,8 +243,8 @@ export class SessionContextClass implements SessionContext {
   }
 
   async setPublicData(data: Record<any, any>) {
-    if (this.userId && data.roles) {
-      await updateAllPublicDataRolesForUser(this.userId, data.roles)
+    if (this.userId) {
+      await updateAllPublicDataRolesForUser(this.userId)
```

Comment:
TODO: We need to decide best approach here. Until now `publicData` is also session specific. The only thing that could change across sessions was `roles`.

Another option is add a config item for which `publicData` fields should be synced across sessions. Actually I kind of like this idea. `roles` and `role` could be set in this list by default.

Reply:
Another great suggestion by @Zeko369: sync all publicData across session by default but add a second config param to `setPublicData` to disable this: `{thisSessionOnly: true}` 

- [ ] Need to check and make sure that `publicData` set during `session.create()` is also synced across sessions.

Reply:
After some discussion I think we should go with this solution:

> Another option is add a config item for which publicData fields should be synced across sessions. Actually I kind of like this idea. roles and role could be set in this list by default.